pkg/plugin/dogu-config/service.go

```diff
@@ -0,0 +1,160 @@
+package dogu_config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/phayes/freeport"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/cloudogu/cesapp-lib/registry"
+	"github.com/cloudogu/kubectl-ces-plugin/pkg/portforward"
+)
+
+func New(doguName, namespace string, restConfig *rest.Config) (*doguConfigService, error) {
+	forwarder, reg, err := createPortForwardAndRegistry(namespace, restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	delegator := newDelegator(doguName, forwarder, reg)
+
+	return &doguConfigService{
+		delegator: delegator,
+	}, nil
+}
+
+func createPortForwardAndRegistry(namespace string, restConfig *rest.Config) (portForwarder, cesRegistry, error) {
+	freePort, err := freeport.GetFreePort()
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not find free port for port-forward: %w", err)
+	}
+
+	forward := portforward.New(restConfig, types.NamespacedName{Namespace: namespace, Name: "etcd-0"}, freePort, 2379)
+
+	endpoint := fmt.Sprintf("http://localhost:%d", freePort)
+	reg, err := registry.New(core.Registry{
+		Type:      "etcd",
+		Endpoints: []string{endpoint},
+	})
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not create etcd registry: %w", err)
+	}
+
+	return forward, reg, nil
+}
+
+type doguConfigService struct {
+	delegator delegator
+}
+
+// Edit opens an interactive dialogue to edit the value of the given key.
+func (s doguConfigService) Edit(registryKey string, deleteOnEmpty bool) error {
+	return s.delegator.Delegate(func(dogu *core.Dogu, editor doguConfigurationEditor) error {
+		matchingFields := matchConfigurationFields(dogu, registryKey)
+		if matchingFields == nil || len(matchingFields) == 0 {
```

Comment:
🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1009: should omit nil check; len() for []github.com/cloudogu/cesapp-lib/core.ConfigurationField is defined as zero